crates/ruff_python_semantic/src/model.rs

```diff
@@ -1498,7 +1498,7 @@ impl<'a> SemanticModel<'a> {
                     source: self.node_id,
                     context: self.execution_context(),
                     exceptions: self.exceptions(),
-                    flags: BindingFlags::empty(),
+                    flags: BindingFlags::GLOBAL,
```

Comment:
I don't think this is the right fix. The binding pushed here is for the variable `_x` in the global scope. But `_x` isn't considered a global-binding in the global scope. It is only when used in a nested scope. For example, the checker explicitly skips over the global handling when in the global scope here

https://github.com/astral-sh/ruff/blob/d4ee6abf4aef3615540a4a84f16edf5e05541622/crates/ruff_linter/src/checkers/ast/mod.rs#L657-L677

But I think you're on the right track. What I find suspicious (and changing it fixes the bug as well) is that `scope` is `self.scope_id` but we add the binding to `self.global_scope_mut`. That's why I think it should actually be `ScopeId::global()` instead. But I'd like a confirmation from @charliermarsh  on this change

Reply:
(Yes, sorry, will take a look.)

Reply:
Hmm... I guess `ScopeId::global()` seems right? We should have decent test coverage here, so if we change that and it passes tests, I'd assume it's correct.

Reply:
Thank you. @Garrett-R do you want to change your PR accordingly?

Reply:
@MichaReiser thank you for that explanation, super helpful! 

Ok, I've updated the PR.